setup.sh

```diff
@@ -338,6 +287,35 @@ fi
 ./compose-dash.sh up -d
 echo "-----------------------------------------"
 
+# Run Local Access mode network scan
+if [ "${config}" == "Local Access" ] && ! grep -qE "^PW_HOST=.+" "${PW_ENV_FILE}"; then
+    echo "Running network scan... (press Ctrl-C to interrupt)"
+    if type winpty > /dev/null 2>&1; then
+        IP=$(netstat -rn | grep "0.0.0.0" | awk '{ print $4 }' | head -1)
+    else
+        IP=$(ip route get 8.8.8.8 | awk '{ print $7 }')
```

Comment:
This doesn't work on my Mac.  I can use `ipconfig getifaddr en0`  or `ifconfig | grep 'inet ' | grep -v '127.0.0.1' | awk '{print $2}'`.

maybe something like this?

```bash
OS=$(uname -s)

# Get local IP based on the operating system
case $OS in
  Linux*) IP=$(ip route get 8.8.8.8 | awk '{ print $7 }') ;;
  Darwin*) IP=$(ifconfig | grep 'inet ' | grep -v '127.0.0.1' | awk '{print $2}') ;;
  CYGWIN*|MINGW32*|MSYS*) IP=$(netstat -rn | grep "0.0.0.0" | awk '{ print $4 }' | head -1) ;;
  *) IP=""
esac

echo "Your OS is ${OS} and your local IP is ${IP}"
```

Reply:
That looks better than what I did. I will incorporate this or similar into the script - feel free to change it though.

I was trying to find a consistent OS independent way of discovering the local IP address in bash.... it seems there isn't one?

Reply:
Sounds good!  Yes, search for one and even asked ChatGPT! 😂  No luck. Worse case, if it gets an empty or wrong, response, the scan won't rung and the user can manually enter the IP during setup.

Reply:
Yep true. When you think something should be simple... lol

To complicate things, small change for this I'll remove the "32" in `CYGWIN*|MINGW32*|MSYS*)` as my Git Bash returns MINGW64 😆

Reply:
It looks good and we are getting close @mcbirse. What left on your list? I'll start testing.